pkg/infrastructure/services/deployer.go

```diff
@@ -222,10 +227,45 @@ func (s *serviceDeployer) Deploy() (reconcileAfter time.Duration, err error) {
 	}
 
 	reconcileAfter, err = s.configureMessaging()
+	if err != nil || reconcileAfter > 0 {
+		return
+	}
+
+	dashboards, err := FetchGrafanaDashboards(s.client, s.instance)
+	if err != nil {
+		return reconciliationPeriodAfterDashboardsError, err
+	}
+
+	reconcileAfter, err = deployGrafanaDashboards(dashboards, s.client, s.getNamespace())
```

Comment:
Please move this to the `s.configureMonitoring()` function, there refactor to call a `Prometheus` and `Grafana` resources.

Reply:
Hi @ricardozanini , what about 
```

func (s *serviceDeployer) configureMonitoring() (time.Duration, error) {
	if failedVerifyAddon, err := configurePrometheus(s.client, s.instance, s.scheme); err != nil {
		return 0, err
	} else if failedVerifyAddon {
		return reconciliationPeriodMonitoringEndpointNotAvailable, nil
	}

	reconcileAfter, err := configureGrafanaDashboards(s.client, s.instance, s.getNamespace())
	if err != nil {
		return reconciliationPeriodAfterDashboardsError, err
	}

	return reconcileAfter, err
}
```
and then create the function `configureGrafanaDashboards` in the `dashboards.go` file, moving there also the existing function `deployGrafanaDashboards`?

Reply:
I pushed a proposal, let me know what do you think. I can revert it just in case :)